src/types/api/response.rs

```diff
@@ -114,3 +117,114 @@ impl LibraryItemsResponse {
         Self(Vec::new())
     }
 }
+
+#[serde_as]
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct SkipGapsResponse {
+    /// Returns the matched attribute: Opensubtitles Hash, File name or season/episode
+    ///
+    /// Primarily used for debugging
+    pub accuracy: String,
+    /// The key of the map is the duration of video
+    /// and the value is the skip gaps - skip history and outro
+    #[serde_as(as = "HashMap<DisplayFromStr, _>")]
+    pub gaps: HashMap<u64, SkipGaps>,
+}
+
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct SkipGaps {
+    #[serde(default)]
+    pub seek_history: Vec<SeekEvent>,
+    #[serde(default)]
+    pub outro: Option<u64>,
+}
+
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct SeekEvent {
+    pub records: u64,
+    #[serde(rename = "seekFrom")]
+    pub from: u64,
+    #[serde(rename = "seekTo")]
+    pub to: u64,
+}
+
+#[cfg(test)]
```

Comment:
Tests should be moved to `unit_tests`

Reply:
We've discussed this with Niki. Adding unit tests in the module is how rust projects test. tests which are integration etc. should be put in `/tests` so we just keep the `unit_tests` for now but this is more idiomatic